persistence/sql/wait_until.go

```diff
@@ -199,6 +202,40 @@ func (p sqlProcessStoreImpl) updateWaitUntilExecution(
 		return nil, err
 	}
 
+	// Step 2 - 2: create timer command tasks
+	var fireTimestamps []int64
+
+	for idx, timerCommand := range request.CommandRequest.TimerCommands {
+		// TODO: define the minimum delay in API
+		if timerCommand.DelayInSeconds < 3 {
```

Comment:
Is it possible to define a constant in the API so that both SDK and server sides can check? Also, should we define a min delay?

Reply:
There are cases where the delay could be 1 or even zero, especially if users call `TimerCommand.byFireTime(...)` because the fireTime is passed from somewhere else. I don't think returning an error here is useful. It would just makes things harder for them.

Cadence/Temporal, even support 0 delay(and there is an optimization to not create timer in server, but I feel like the optimization is a bit complicated to do for now). 

We can do ensure the delay is not <0. Make it 0 when it's <0. But probaly doing in  API layer is better to keep the persistence layer a bit cleaner (our persistence layer is growing so fast )

Reply:
delete this part and add logic to deal with negative delay.